### PR TITLE
Bug 2042516: Fixes for shared to local gateway migration

### PR DESCRIPTION
**Pull request by trozet**
Changes-Include:

updating service route correctly
removing egress firewalls from join switch on sgw -> local

### CODE REVIEW

File: go-controller/pkg/ovn/egressfirewall.go [hunk 1]
```diff
@@ -138,15 +146,50 @@ func (oc *Controller) syncEgressFirewall(egressFirwalls []interface{}) {
...
+						if err != nil {
+							klog.Errorf("Unable to remove egress firewall acl: %s on %s, cannot cleanup old stale"+
+								" data, stderr: %s, err: %v", egressFirewallACLID, types.OVNJoinSwitch, stderr, err)
 						}
 					}
 				}
 			}
 		}
 	}
+
```
Comment (tssurya) [inline go-controller/pkg/ovn/egressfirewall.go:190]: are we missing something from 4.9 here: https://github.com/openshift/ovn-kubernetes/commit/50ea1092e4b9b05815e4b401dd6c10eb534c452a#diff-3be8fe02b70b5dfce92915a60b8901d324a5d2da49058dace0d8437dfc153c1bL153?
I think this is the only conflict, but looks ok.

**Review by tssurya — COMMENTED** [review]
code looks ok, mostly clean pick from 4.9 except for a small conflict.
let's wait for the migration job o pass.

File: go-controller/pkg/ovn/egressfirewall.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/ovn/egressfirewall.go:190]: yeah we are missing https://github.com/openshift/ovn-kubernetes/pull/592/commits/e6953481902e4397448b500466a60c14fdf941ed in 4.8, but it is in 4.7

**Comment by tssurya** [comment]
/lgtm
migration job succeeded :D woot!

**Comment by trozet** [comment]
@anuragthehatter PTAL

**Pull request merged**